rust/lanzatool/src/gc.rs

```diff
@@ -35,10 +35,30 @@ impl Roots {
     }
 
     pub fn collect_garbage(&self, directory: impl AsRef<Path>) -> Result<()> {
+        self.collect_garbage_with_filter(directory, |_| true)
+    }
+
+    /// Collect garbage with an additional filter.
+    ///
+    /// The filter function takes a &Path and returns a bool. The paths for which the filter
+    /// function returns true are considered for garbage collection. This means that _only_ files
+    /// that are unused AND for which the filter function returns true are deleted.
+    pub fn collect_garbage_with_filter<P>(
+        &self,
+        directory: impl AsRef<Path>,
+        mut predicate: P,
+    ) -> Result<()>
+    where
+        P: FnMut(&Path) -> bool,
+    {
         // Find all the paths not used anymore.
         let entries_not_in_use = WalkDir::new(directory.as_ref())
             .into_iter()
-            .filter(|e| !self.in_use(e.as_ref().ok()));
+            .filter(|e| !self.in_use(e.as_ref().ok()))
```

Comment:
I sure hope that Rust performs stream fusion

Reply:
No idea, but I trust rustc will do the right thing™

Reply:
inchallah as we say